gradle/project.gradle.kts

```diff
@@ -36,9 +31,15 @@ kotlin {
     experimental.coroutines = Coroutines.ENABLE
 }
 
-allprojects {
-    buildDir = File("Z://build")
+/** working directory of the application */
+val workingDir = file("working dir")
+val kotlinVersion: String by extra {
+    buildscript.configurations["classpath"]
+            .resolvedConfiguration.firstLevelModuleDependencies
+            .find { it.moduleName=="org.jetbrains.kotlin.jvm.gradle.plugin" }!!.moduleVersion
```

Comment:
This is useless in my opinion, as version string provided by developer in constant is enough.

Reply:
the advantage is that this way, the kotlin version is only defined in the plugins block, and this variable extracts the kotlinVersion from the plugin. Moveing it to the top and defining is as a constant would duplicate the work.

Reply:
I do not understand, would not a constant do the same job, but avoid the fragile and painfully long lookup?

Reply:
see above, the kotlinVersion is already defined in the plugins block: `kotlin("jvm") version "1.2.21"`

this is only here to make the version accessible in the whole script

Reply:
Ok